app/src/org/commcare/models/database/InterruptedFormState.java

```diff
@@ -17,10 +18,12 @@ public class InterruptedFormState implements Externalizable {
 
     int sessionStateDescriptorId;
     FormIndex formIndex;
+    private boolean interruptedDueToSessionExpiration = false;
```

Comment:
possibly have it as enum - `InterruptionCause` with values `SESSION_EXPIRATION` and `PROCESS_KILL`

Reply:
@shubham1g5 I entertained this idea too, the issue is that there is no implicit call for `PROCESS_KILL`, we kind of do preventive saves. so it would be inconsistent with the triggering process. Should we still call is `PROCESS_KILL` or `PREVENTIVE_SAVE`, I'm ok with either.

Reply:
Makes sense, I think fine to leave it as it is.